ocs_ci/ocs/managedservice.py

```diff
@@ -93,3 +96,72 @@ def get_consumer_names():
     )
     consumer_yamls = consumer.get().get("items")
     return [consumer["metadata"]["name"] for consumer in consumer_yamls]
+
+
+def patch_consumer_toolbox(ceph_admin_key=None):
+    """
+    Patch the rook-ceph-tools deployment with ceph.admin key. Applicable for MS platform only to enable rook-ceph-tools
+    to run ceph commands.
+
+    Args:
+        ceph_admin_key (str): The ceph admin key which should be used to patch rook-ceph-tools deployment on consumer
+
+    """
+
+    # Get the admin key if available
+    ceph_admin_key = (
+        ceph_admin_key
+        or os.environ.get("CEPHADMINKEY")
+        or config.AUTH.get("external", {}).get("ceph_admin_key")
+    )
+
+    if not ceph_admin_key:
+        # TODO: Get the key from provider rook-ceph-tools pod after implementing multicluster deployment
+        logger.warning(
+            "Ceph admin key not found to patch rook-ceph-tools deployment on consumer with ceph.admin key. "
+            "Skipping the step."
+        )
+        return
+
+    consumer_tools_pod = get_ceph_tools_pod()
+
+    # Check whether ceph command is working on tools pod. Patch is needed only if the error is "RADOS permission error"
+    try:
+        consumer_tools_pod.exec_ceph_cmd("ceph health")
```

Comment:
Don't you want to stop the function's execution if the ceph command succeeds?

Reply:
@ebondare Yes, that is correct. Fixed now.